src/world/action/walk.c

```diff
@@ -1,13 +1,125 @@
 #include "common.h"
+#include "world/actions.h"
+
+s32 D_802B6910_E23FF0[] = {0x000A0002, 0x000A002B, 0x000A002D, 0x000A002F, 0x000A0031, 0x000A0033, 0x000A0035, 0x000A0037, 0x000A0039, 0x000A003B, 0x000A003D, 0x000A003F,
+                           0x000A0041, 0x000A0043, 0x000A0045, 0x000A0047, 0x000A0049, 0x000A004B, 0x000A004D, 0x00000000 };
 
 // walk
 INCLUDE_ASM(void, "world/action/walk", func_802B6000_E236E0, void);
 
 // run
-INCLUDE_ASM(void, "world/action/walk", func_802B6288_E23968, void);
+void func_802B6288_E23968() {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PlayerData* playerData = &gPlayerData;
+    f32 sp14;
+    f32 sp10;
+    s32 temp_v1;
+    s32 phi_a0;
+    f32 phi_f2;
+    s32 phi_s3;
+    s32 xAxis;
+    s32 yAxis;
+
+    phi_s3 = 0;
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS) {
+        func_802B6738_E23E18();
+        return;
+    }
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= ~0x80880000;
+        D_8010C980 = 0;
+        playerStatus->unk_60 = 0;
+        phi_s3 = 1;
+
+        if (!(playerStatus->flags & 0x00004000)) {
+            playerStatus->currentSpeed = playerStatus->runSpeed;
+        }
+        if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+            phi_a0 = 0x90003;
+        } else {
+            if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_HOLDING_ITEM)) {
+                phi_a0 = 0x10005;
+            } else {
+                phi_a0 = 0x60002;
+            }
+        }
+        suggest_player_anim_clearUnkFlag(phi_a0);
+    }
+
+    if (playerStatus->flags & 0x00004000) {
+        playerStatus->targetYaw = playerStatus->heading;
+        try_player_footstep_sounds(4);
+        return;
+    }
 
-// same as func_802B6198_E24768 but requires data stuff
-INCLUDE_ASM(void, "world/action/walk", func_802B6550_E23C30, void);
+    D_8010C980++;
+    phi_f2 = 1.0f;
+
+    if (playerStatus->animFlags & 0x10000) {
+        phi_f2 = 1.5f;
+    }
+
+    playerStatus->currentSpeed = playerStatus->runSpeed * phi_f2;
+    player_input_to_move_vector(&sp10, &sp14);
+    phys_update_interact_collider();
+    if (check_input_jump() == FALSE) {
+        if (phi_s3 == 0) {
+            if (check_input_hammer()) {
+                return;
+            }
+        }
+
+        player_input_to_move_vector(&sp10, &sp14);
```

Comment:
Not required, but you could name sp10 and sp14 `moveX` and `moveY` due to this call and declare them in this block since they're only(?) used here

Reply:
I can change the names but I can't move the declaration as I have to call player_input_to_move_vector at line 64 as well

Reply:
nw, i missed that